src/tonic/api/collections_api.rs

```diff
@@ -0,0 +1,136 @@
+use tonic::{Request, Response, Status};
+
+use crate::tonic::proto::collections_server::Collections;
+use crate::tonic::proto::{
+    CreateCollection, HnswConfigDiff, OptimizersConfigDiff, StorageOperations,
+    UpdateCollectionsResponse, WalConfigDiff,
+};
+use num_traits::FromPrimitive;
+use std::convert::TryFrom;
+use std::sync::Arc;
+use std::time::Instant;
+use storage::content_manager::errors::StorageError;
+use storage::content_manager::toc::TableOfContent;
+
+pub struct CollectionsService {
+    toc: Arc<TableOfContent>,
+}
+
+impl CollectionsService {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self { toc }
+    }
+}
+
+#[tonic::async_trait]
+impl Collections for CollectionsService {
+    async fn update_collections(
+        &self,
+        request: Request<StorageOperations>,
+    ) -> Result<Response<UpdateCollectionsResponse>, Status> {
+        let operations = storage::content_manager::storage_ops::StorageOperations::try_from(
+            request.into_inner(),
+        )?;
+        let timing = Instant::now();
+        let result = self.toc.perform_collection_operation(operations).await;
+
+        let response = UpdateCollectionsResponse::from((timing, result));
+        Ok(Response::new(response))
+    }
+}
+
+impl TryFrom<StorageOperations> for storage::content_manager::storage_ops::StorageOperations {
+    type Error = Status;
+
+    fn try_from(value: StorageOperations) -> Result<Self, Self::Error> {
+        if let Some(create_collection) = value.create_collection {
+            Ok(Self::try_from(create_collection)?)
+            // } else if let Some(update_collection) = self.update_collection {
```

Comment:
Will do it in the next PR. Would like to get a feedback about current approach first